orders/views.py

```diff
@@ -86,10 +99,99 @@ def post(self, request):
+
+
+
+
+class CartView(View):
+    @login
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+
+            if Cart.objects.filter(Q(item_id = data['item_id']) & Q(member_id=request.user.id)).exists():
+                return JsonResponse({"MESSAGE":"CART_EXIST"}, status=400)
+            
+            if data['quantity'] > Item.objects.get(id = data['item_id']).stock:
+                return JsonResponse({"MESSAGE":"NO_STOCK"}, status=400)
+                
+            Cart.objects.create(
+                item_id=data['item_id'], 
+                member_id = request.user.id, 
+                quantity = data['quantity']
+            )
+            return JsonResponse({"MESSAGE" : "SUCCESS"}, status=201)
+
+        except KeyError:
+            return JsonResponse({"MESSAGE": "KEY_ERROR"}, status=400)
+
+    @login
+    def get(self, request):
+
+        if not Cart.objects.filter(member_id=request.user.id).exists():
+                return JsonResponse({"MESSAGE":"EMPTY_CART"}, status=400)
+
+        results = [
+            {
+            "cart_id"  : cart.id,
+            "item_id"  : cart.item.id,
+            "quantity" : cart.quantity,
+            "name"     : cart.item.name,
+            "image"    : cart.item.image_set.get(main=1).image_url,
+            "price"    : cart.item.price,
+            "discount" : cart.item.discount
+            } 
+        for cart in Cart.objects.filter(member_id=request.user.id)]
+
+        return JsonResponse({"RESULT" : results}, status=201)
+
+    @login
+    def patch(self, request):
+        is_cart = request.GET.get("is_cart", None)
+        item_id = request.GET.get("item_id", None)
+        quantities = request.GET.get("quantity", None)
+        cart_id = request.GET.get("cart_id", None)
+
+        if not is_cart:
+            cart = Cart.objects.get(Q(item_id = item_id) & Q(member_id=request.user.id))
+            if cart.item.stock < cart.quantity + int(quantities):
+                return JsonResponse({"MESSAGE":"NO_STOCK"}, status=400)
+
+            cart.quantity = F('quantity') + int(quantities)
+            cart.save()
+            return JsonResponse({"MESSAGE" : "SUCCESS"}, status=200)
+
+        cart = Cart.objects.get(id=cart_id)
+        if cart.item.stock < int(quantities):
+            return JsonResponse({"MESSAGE":"NO_STOCK"}, status=400)
+        
+        cart.quantity = int(quantities)
+        cart.save()
+        
+        return JsonResponse({"MESSAGE" : "SUCCESS"}, status=200)
+
+    @login
+    def delete(self, request):
+        is_all = request.GET.get("is_all", None)
+        cart_id = request.GET.get("cart_id", None)
+
+        if not is_all:
+            Cart.objects.get(id = cart_id).delete()
+            return JsonResponse({"MESSAGE" : "DELETE"}, status=200)
+
+        Cart.objects.filter(member_id = request.user.id).delete()
+        
+        return JsonResponse({"MESSAGE" : "SUCCESS"}, status=200)
```

Comment:
cart를 하나씩 지우는 방식을 구현 하신다면, `DELETE /carts/<int:cart_id>` 형태로 구현해야 하지만, 위 케이스처럼 전체 상품도 삭제해야되는 경우는, `DELETE /carts?ids=[1,2,3,5]`로 받아서 삭제하도록 구현해주세요. 그래야만 한개를 삭제할 수도, 여러개를 삭제할 수도 있습니다.

```suggestion
    @login
    def delete(self, request):
        is_all = request.GET.get("is_all", None)
        cart_id = request.GET.get("cart_id", None)

        if not is_all:
            Cart.objects.get(id = cart_id).delete()
            return JsonResponse({"MESSAGE" : "DELETE"}, status=200)

        Cart.objects.filter(member_id = request.user.id).delete()
        
        return JsonResponse({"MESSAGE" : "SUCCESS"}, status=200)
```

Reply:
추가로 자원을 삭제했으면 status 코드는 204를 반환합니다.
```python
        return JsonResponse({"MESSAGE" : "NO_CONTENT"}, status=204)
```